src/session/session_api.c

```diff
@@ -804,27 +804,33 @@ __wt_session_drop(WT_SESSION_IMPL *session, const char *uri, const char *cfg[])
 	WT_RET(__wt_config_gets_def(session, cfg, "checkpoint_wait", 1, &cval));
 	checkpoint_wait = cval.val != 0;
 	WT_RET(__wt_config_gets_def(session, cfg, "lock_wait", 1, &cval));
-	lock_wait = cval.val != 0 || F_ISSET(session, WT_SESSION_LOCK_NO_WAIT);
-
-	if (!lock_wait)
-		F_SET(session, WT_SESSION_LOCK_NO_WAIT);
```

Comment:
> The variable was badly named, but that was determining if `WT_SESSION_LOCK_NO_WAIT` was already set on a session before trying to set it again (and later clear it).

OK, good -- I don't think there's a path where that can happen, but obviously we don't have to worry about it unless we revert back to using a `WT_SESSION_IMPL` flag.